akka-actor-tests/src/test/scala/akka/util/ManifestInfoSpec.scala

```diff
@@ -18,6 +18,19 @@ class ManifestInfoSpec extends AkkaSpec {
       ManifestInfo.checkSameVersion("Akka", allModules, versions) shouldBe Some(
         "You are using version 2.6.4 of Akka, but it appears you (perhaps indirectly) also depend on older versions of related artifacts. " +
         "You can solve this by adding an explicit dependency on version 2.6.4 of the [akka-persistence, akka-cluster] artifacts to your project. " +
+        "Here's a complete collection of detected artifacts: (2.6.4, [akka-actor]), (2.5.3, [akka-persistence, akka-cluster]). " +
+        "See also: https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed")
+    }
+
+    "support dynver" in {
+      val versions = Map(
+        "akka-actor" -> new ManifestInfo.Version("2.6.4"),
+        "akka-persistence" -> new ManifestInfo.Version("2.6.4+10-abababef"))
+      val allModules = List("akka-actor", "akka-persistence")
+      ManifestInfo.checkSameVersion("Akka", allModules, versions) shouldBe Some(
+        "You are using version 2.6.4+10-abababef of Akka, but it appears you (perhaps indirectly) also depend on older versions of related artifacts. " +
+        "You can solve this by adding an explicit dependency on version 2.6.4+10-abababef of the [akka-actor] artifacts to your project. " +
+        "Here's a complete collection of detected artifacts: (2.6.4, [akka-actor]), (2.6.4+10-abababef, [akka-persistence]). " +
         "See also: https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed")
```

Comment:
I had some unexpected output on a test run in a separate project and, for a moment, suspected dynver was not supported.

I've left the test I added but I don't think it's a necessary test and would happily remove it.

Reply:
Heh! turns out the dynver test caught an edge case as I was introducing support to sort the output.